src/main/java/org/jitsi/videobridge/OctoChannel.java

```diff
@@ -295,20 +320,47 @@ public boolean accept(DatagramPacket p)
             MediaType packetMediaType
                 = OctoPacket.readMediaType(
                         p.getData(), p.getOffset(), p.getLength());
-            if (packetMediaType != mediaType)
-            {
-                return false;
-            }
 
-            // The RTP/RTCP packet is preceded by the fixed length Octo header
-            boolean packetIsRtcp
-                = RTCPUtils.isRtcp(
+            if (mediaType.equals(packetMediaType))
+            {
+                // The RTP/RTCP packet is preceded by the fixed length Octo
+                // header.
+                boolean packetIsRtcp
+                    = RTCPUtils.isRtcp(
                         p.getData(),
                         p.getOffset() + OctoPacket.OCTO_HEADER_LENGTH,
                         p.getLength() - OctoPacket.OCTO_HEADER_LENGTH);
 
-            // Note that the rtp socket gets all non-rtcp
-            return rtcp == packetIsRtcp;
+                // Note that the rtp socket gets all non-rtcp packets.
+                return rtcp == packetIsRtcp;
+            }
+
+            if (MediaType.DATA.equals(packetMediaType) && handleData)
+            {
+                // In this case we're going to return 'false' anyway, because
+                // we already read the contents and we don't want the packet to
+                // be accepted by libjitsi.
+                handleDataPacket(p);
+            }
+
+            return false;
         }
     }
+
+    /**
+     * TODO
+     */
+    private void handleDataPacket(DatagramPacket p)
```

Comment:
what does a `data` packet refer to here?  data from a client's datachannel?

Reply:
Yes. I'll probably rename this when I get it done, because it's confusing (suggestions are welcome)